File: app/src/main/java/com/websarva/wings/android/zuboradiary/data/preferences/UserPreferencesRepository.java
package com.websarva.wings.android.zuboradiary.data.preferences;

import androidx.datastore.preferences.core.Preferences;

import java.util.Objects;

import io.reactivex.rxjava3.core.Flowable;
import io.reactivex.rxjava3.core.Single;

public class UserPreferencesRepository {

    private final UserPreferences userPreferences;

    public UserPreferencesRepository(UserPreferences userPreferences) {
        Objects.requireNonNull(userPreferences);

        this.userPreferences = userPreferences;
    }

    public Flowable<ThemeColorPreferenceValue> loadThemeColorSettingValue() {
        return userPreferences.loadThemeColorPreferenceValue();
    }

    public Single<Preferences> saveThemeColorPreferenceValue(ThemeColorPreferenceValue value) {
        return userPreferences.saveThemeColorPreferenceValue(value);
    }

    public Flowable<CalendarStartDayOfWeekPreferenceValue> loadCalendarStartDayOfWeekPreferenceValue() {
        return userPreferences.loadCalendarStartDayOfWeekPreferenceValue();
    }

    public Single<Preferences> saveCalendarStartDayOfWeekPreferenceValue(CalendarStartDayOfWeekPreferenceValue value) {
        return userPreferences.saveCalendarStartDayOfWeekPreferenceValue(value);
    }

    public Flowable<ReminderNotificationPreferenceValue> loadReminderNotificationPreferenceValue() {
        return userPreferences.loadReminderNotificationPreferenceValue();
    }

    public Single<Preferences> saveReminderNotificationPreferenceValue(ReminderNotificationPreferenceValue value) {
        return userPreferences.saveReminderNotificationPreferenceValue(value);
    }

    public Flowable<PassCodeLockPreferenceValue> loadPasscodeLockPreferenceValue() {
        return userPreferences.loadPasscodeLockPreferenceValue();
    }

    public Single<Preferences> savePasscodeLockPreferenceValue(PassCodeLockPreferenceValue value) {
        return userPreferences.savePasscodeLockPreferenceValue(value);
    }

    public Flowable<WeatherInfoAcquisitionPreferenceValue> loadWeatherInfoAcquisitionPreferenceValue() {
        return userPreferences.loadWeatherInfoAcquisitionPreferenceValue();
    }

    public Single<Preferences> saveWeatherInfoAcquisitionPreferenceValue(WeatherInfoAcquisitionPreferenceValue value) {
        return userPreferences.saveWeatherInfoAcquisitionPreferenceValue(value);
    }
}

File: app/src/main/java/com/websarva/wings/android/zuboradiary/data/preferences/UserPreferencesRepository.kt
package com.websarva.wings.android.zuboradiary.data.preferences

import androidx.datastore.core.IOException
import kotlinx.coroutines.flow.Flow

class UserPreferencesRepository(private val userPreferences: UserPreferences) {

    @Throws(Throwable::class)
    fun loadThemeColorPreference(): Flow<ThemeColorPreference> {
        return userPreferences.loadThemeColorPreference()
    }

    @Throws(
        IOException::class,
        Exception::class
    )
    suspend fun saveThemeColorPreference(preference: ThemeColorPreference) {
        userPreferences.saveThemeColorPreference(preference)
    }

    @Throws(Throwable::class)
    fun loadCalendarStartDayOfWeekPreference(): Flow<CalendarStartDayOfWeekPreference> {
        return userPreferences.loadCalendarStartDayOfWeekPreference()
    }

    @Throws(
        IOException::class,
        Exception::class
    )
    suspend fun saveCalendarStartDayOfWeekPreference(preference: CalendarStartDayOfWeekPreference) {
        return userPreferences.saveCalendarStartDayOfWeekPreference(preference)
    }

    @Throws(Throwable::class)
    fun loadReminderNotificationPreference(): Flow<ReminderNotificationPreference> {
        return userPreferences.loadReminderNotificationPreference()
    }

    @Throws(
        IOException::class,
        Exception::class
    )
    suspend fun saveReminderNotificationPreference(preference: ReminderNotificationPreference) {
        return userPreferences.saveReminderNotificationPreference(preference)
    }

    @Throws(Throwable::class)
    fun loadPasscodeLockPreference(): Flow<PassCodeLockPreference> {
        return userPreferences.loadPasscodeLockPreference()
    }

    @Throws(
        IOException::class,
        Exception::class
    )
    suspend fun savePasscodeLockPreference(preference: PassCodeLockPreference) {
        return userPreferences.savePasscodeLockPreference(preference)
    }

    @Throws(Throwable::class)
    fun loadWeatherInfoAcquisitionPreference(): Flow<WeatherInfoAcquisitionPreference> {
        return userPreferences.loadWeatherInfoAcquisitionPreference()
    }

    @Throws(
        IOException::class,
        Exception::class
    )
    suspend fun saveWeatherInfoAcquisitionPreference(preference: WeatherInfoAcquisitionPreference) {
        return userPreferences.saveWeatherInfoAcquisitionPreference(preference)
    }

    @Throws(
        IOException::class,
        Exception::class
    )
    suspend fun initializeAllPreferences() {
        return userPreferences.initializeAllPreferences()
    }
}
